File: 프로그래머스/lv1/92334. 신고 결과 받기/신고 결과 받기.js
function solution(id_list, report, k) {
  const target = {};
  const targetCount = {};

  const reportDeduplication = [...new Set(report)];

  for (const item of reportDeduplication) {
    const splitedItem = item.split(" ");
    if (targetCount[splitedItem[1]]) targetCount[splitedItem[1]]++;
    else targetCount[splitedItem[1]] = 1;

    if (target[splitedItem[0]]) target[splitedItem[0]].push(splitedItem[1]);
    else target[splitedItem[0]] = [splitedItem[1]];
  }

  const answer = [];

  for (const item of id_list) {
    let count = 0;
    if (target[item]) {
      for (const targetItem of target[item]) {
        if (targetCount[targetItem] >= k) {
          count++;
        }
      }
      answer.push(count);
    } else {
      answer.push(0);
    }
  }

  return answer;
}

File: 프로그래머스/lv1/92334. 신고 결과 받기/신고 결과 받기.py
def solution(id_list, report, k):
    answer = []
    report = list(set(report))
    # id를 키로하여 신고한 아이디를 값으로 가지는 오브젝트
    target_obj = {}
    # id별 신고당한 횟수를 카운팅하는 오브젝트
    count_obj = {}
    
    for item in report:
        me, you = item.split(" ")
        
        if me in target_obj:
            target_obj[me].append(you)
        else:
            target_obj[me] = [you]
            
        if you in count_obj:
            count_obj[you] += 1
        else:
            count_obj[you] = 1
    
    reported_users = [i for i in count_obj if count_obj[i] >= k]
    # print(target_obj, count_obj)
    # print(reported_users)
    for i in id_list:
        tmp = 0
        if i in target_obj:
            for j in target_obj[i]:
                if j in reported_users:
                    tmp += 1
            answer.append(tmp)
        else:
            answer.append(tmp)
    return answer